fastlane/lanes/release.rb

```diff
@@ -379,53 +391,90 @@
   # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
   #
   desc 'Trigger the final release build on CI'
-  lane :finalize_release do |options|
+  lane :finalize_release do |skip_confirm: false|
     UI.user_error!('To finalize a hotfix, please use the finalize_hotfix_release lane instead') if ios_current_branch_is_hotfix
 
-    ensure_git_branch_is_release_branch
-
-    # Verify that there's nothing in progress in the working copy
+    ensure_git_branch_is_release_branch!
     ensure_git_status_clean
 
-    skip_user_confirmation = options[:skip_confirm]
-
     UI.important("Finalizing release: #{release_version_current}")
-    UI.user_error!('Aborted by user request') unless skip_user_confirmation || UI.confirm('Do you want to continue?')
-
-    git_pull
+    UI.user_error!('Aborted by user request') unless skip_confirm || UI.confirm('Do you want to continue?')
 
-    check_all_translations(interactive: skip_user_confirmation == false)
+    check_all_translations(interactive: skip_confirm == false)
 
     download_localized_strings_and_metadata(options)
-    lint_localizations(allow_retry: skip_user_confirmation == false)
+    lint_localizations(allow_retry: skip_confirm == false)
 
     bump_build_codes
 
-    unless skip_user_confirmation || UI.confirm('Ready to push changes to remote and trigger the release build?')
-      UI.message("Terminating as requested. Don't forget to run the remainder of this automation manually.")
-      next
+    unless skip_confirm || UI.confirm('Ready to push changes to remote and trigger the release build?')
+      UI.user_error!("Terminating as requested. Don't forget to run the remainder of this automation manually.")
     end
 
     push_to_git_remote(tags: false)
 
     version = release_version_current
-    remove_branch_protection(repository: GITHUB_REPO, branch: release_branch_name)
-    set_milestone_frozen_marker(repository: GITHUB_REPO, milestone: version, freeze: false)
-    create_new_milestone(repository: GITHUB_REPO)
-    close_milestone(repository: GITHUB_REPO, milestone: version)
 
     trigger_release_build
 
-    pr_url = create_release_management_pull_request(
-      release_version: release_version_next,
-      base_branch: DEFAULT_BRANCH,
-      title: "Merge #{version} release finalization"
-    )
-
+    pr_url = create_backmerge_pr
     message = <<~MESSAGE
       Release successfully finalized. Next, review and merge the [integration PR](#{pr_url}).
     MESSAGE
     buildkite_annotate(context: 'finalization-completed', style: 'success', message: message) if is_ci
+
+    remove_branch_protection(repository: GITHUB_REPO, branch: release_branch_name)
```

Comment:
~~Should the step to remove branch protection be moved to the `publish_release` lane?~~

Update: I see the method to delete the branch below removes the branch protection, so I'd say we can remove this altogether

```suggestion
```

Reply:
Ah, nice catch! I believe this is a leftover from when apps didn't have a release_publish lane.